apps/router-e2e/__e2e__/dom-components/components/03-local-asset.tsx

```diff
@@ -1,7 +1,19 @@
 'use dom';
 
-import { Image } from 'react-native';
+import { Image, View } from 'react-native';
 
-export default function Page() {
-  return <Image source={require('../../../assets/icon.png')} />;
+export default function Page(props: import('expo/dom').DOMProps) {
```

Comment:
Upon further consideration, I think we should avoid this for now. It's nice to just use types as-is and not requiring a utility type to make a dom component. This also makes migration a bit weirder since on web you can pass any type you want to a dom component. Would prefer if we keep all magic props behind `dom` (maybe even renaming to something like `$dom` (assuming React and associated dev tooling allows this)).

Reply:
give another try for using `withDOMProps()` HOC. we need this HOC only from native callsites. change at https://github.com/expo/expo/pull/31103/commits/e6039c6986810352788451e3446d5dbbd51af782